pkg/utils/db.go

```diff
@@ -341,39 +349,32 @@ func GetMariaDBUUID(db *sql.DB) (string, error) {
 	return fmt.Sprintf("%d%s%d", domainID, domainServerIDSeparator, serverID), nil
 }
 
-// GetSQLMode returns sql_mode.
-func GetSQLMode(db *sql.DB) (tmysql.SQLMode, error) {
+// GetParser gets a parser for sql.DB which maybe enabled `ANSI_QUOTES` sql_mode
+func GetParser(db *sql.DB) (*parser.Parser, error) {
 	sqlMode, err := GetGlobalVariable(db, "sql_mode")
```

Comment:
Session here ditto

Reply:
changed in https://github.com/pingcap/dm/pull/889/commits/104f5df733698d825843bb8808157ba646c8f306

Reply:
Have not change?

Reply:
want to explain that `(*UpStreamConn).getParser` will use `GetParserForConn`, which further uses session variables so could reflect variables set though DSN. And `(*UpStreamConn).getParser` is used in most unit, in fact `GetParser` is only used in relay unit and tests 😿 

will change it later

Reply:
https://github.com/pingcap/dm/pull/889/commits/02df4e3a36cf0b78b0f51179eb6e5c3f72692a2d